core/domain/exp_domain.py

```diff
@@ -3546,9 +3546,74 @@ def _convert_states_v22_dict_to_v23_dict(cls, states_dict):
                 state_dict, html_cleaner.add_caption_attr_to_image)
         return states_dict
 
+    @classmethod
+    def _convert_states_v23_dict_to_v24_dict(cls, states_dict, exp_id):
+        """Converts from version 23 to 24. Version 24 adds the dimensions of
+        images in the oppia-noninteractive-image tags.
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for key, state_dict in states_dict.iteritems():
```

Comment:
All this stuff is too complicated and we've had similar functions that iterate through all HTML fields in an exploration before. Find those and reuse them, or extract them out into common functions if needed. Don't duplicate code.

Reply:
Done!